ios/RNPdftron.m

```diff
@@ -109,6 +109,30 @@ - (void)setPassword:(NSString *)password onPDFDoc:(PTPDFDoc *)pdfDoc
     }
 }
 
+RCT_EXPORT_METHOD(pdfFromOfficeTemplate:(NSString *)docxPath json:(NSDictionary *)json resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+    @try {
+        PTPDFDoc* pdfDoc = [[PTPDFDoc alloc] init];
+        NSData *jsonData = [NSJSONSerialization dataWithJSONObject:json options:0 error:nil];
+        NSString *jsonString = [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
+        PTOfficeToPDFOptions* options = [[PTOfficeToPDFOptions alloc] init];
+        [options SetTemplateParamsJson:jsonString];
+        [PTConvert OfficeToPDF:pdfDoc in_filename:docxPath options:options];
+        
+        NSString* fileName = [[NSUUID UUID].UUIDString stringByAppendingString:@".pdf"];
```

Comment:
You can replace this with

https://developer.apple.com/documentation/foundation/nsstring/1412501-stringbyappendingpathextension